src/writesbml.jl

```diff
@@ -82,63 +76,66 @@ function add_rule(model::VPtr, r::Union{AssignmentRule,RateRule})
 end
 
 function add_unit_definition(model::VPtr, id::String, units::UnitDefinition)
     unit_definition = ccall(sbml(:Model_createUnitDefinition), VPtr, (VPtr,), model)
-    ccall(sbml(:UnitDefinition_setId), Cint, (VPtr, Cstring), unit_definition, id)
-    isnothing(units.name) || ccall(
-        sbml(:UnitDefinition_setName),
-        Cint,
-        (VPtr, Cstring),
-        unit_definition,
-        units.name,
-    )
+    set_string!(unit_definition, :UnitDefinition_setId, id)
+    set_string!(unit_definition, :UnitDefinition_setName, units.name)
     for unit in units.unit_parts
         unit_ptr = ccall(sbml(:UnitDefinition_createUnit), VPtr, (VPtr,), unit_definition)
         unit_kind = ccall(sbml(:UnitKind_forName), Cint, (Cstring,), unit.kind)
-        ccall(sbml(:Unit_setKind), Cint, (VPtr, Cint), unit_ptr, unit_kind)
-        ccall(sbml(:Unit_setScale), Cint, (VPtr, Cint), unit_ptr, unit.scale)
-        ccall(sbml(:Unit_setExponent), Cint, (VPtr, Cint), unit_ptr, unit.exponent)
-        ccall(sbml(:Unit_setMultiplier), Cint, (VPtr, Cdouble), unit_ptr, unit.multiplier)
+        set_int!(unit_ptr, :Unit_setKind, unit_kind)
+        set_int!(unit_ptr, :Unit_setScale, unit.scale)
+        set_int!(unit_ptr, :Unit_setExponent, unit.exponent)
+        set_double!(unit_ptr, :Unit_setMultiplier, unit.multiplier)
     end
 end
 
 function set_parameter_ptr!(parameter_ptr::VPtr, id::String, parameter::Parameter)::VPtr
-    ccall(sbml(:Parameter_setId), Cint, (VPtr, Cstring), parameter_ptr, id)
-    isnothing(parameter.name) || ccall(
-        sbml(:Parameter_setName),
-        Cint,
-        (VPtr, Cstring),
-        parameter_ptr,
-        parameter.name,
-    )
-    isnothing(parameter.value) || ccall(
-        sbml(:Parameter_setValue),
-        Cint,
-        (VPtr, Cdouble),
-        parameter_ptr,
-        parameter.value,
-    )
-    isnothing(parameter.units) || ccall(
-        sbml(:Parameter_setUnits),
-        Cint,
-        (VPtr, Cstring),
-        parameter_ptr,
-        parameter.units,
-    )
-    isnothing(parameter.constant) || ccall(
-        sbml(:Parameter_setConstant),
-        Cint,
-        (VPtr, Cint),
-        parameter_ptr,
-        Cint(parameter.constant),
-    )
+    set_string!(parameter_ptr, :Parameter_setId, id)
+    set_string!(parameter_ptr, :Parameter_setName, parameter.name)
+    set_double!(parameter_ptr, :Parameter_setValue, parameter.value)
+    set_string!(parameter_ptr, :Parameter_setUnits, parameter.units)
+    set_bool!(parameter_ptr, :Parameter_setConstant, parameter.constant)
+    set_sbo_term!(parameter_ptr, parameter.sbo)
     return parameter_ptr
 end
 
+function set_string!(ptr::VPtr, fn_sym::Symbol, x::Maybe{String})
+    isnothing(x) ||
+        ccall(sbml(fn_sym), Cint, (VPtr, Cstring), ptr, x) == 0 ||
+        error("$fn_sym failed for value `$x' !")
+end
+
+function set_int!(ptr::VPtr, fn_sym::Symbol, x::Maybe{I}) where {I<:Integer}
```

Comment:
If you don't need `I` in the body of the function you can also do `Maybe{<:Integer}`